### PR TITLE
Update Фугапедия presence

### PR DESCRIPTION
**Pull request by Syjalo**
Added:
- Some new cases
- 3 new settings
preview1
preview2

### CODE REVIEW

**Comment by Syjalo** [comment]
I'll upload screenshots later.

**Review by ImRodry — COMMENTED** [review]
So if privacyMode is set to false do you just... not display anything at all? That kinda defeats the purpose of the presence imo

**Comment by Syjalo** [comment]
> 

No, if privacyMode is set to false I display what they do: reading article, reading book, looking for article, etc. And some additional information: for articles it's article title, for user it's username + name, etc.
If privacyMode is set to true I display only what what they do.

**Comment by Syjalo** [comment]
For example:

> privacyMode: `false`
> Reading article
> {articleTitle}
> <button>Go to the article</button>

poff1

> privacyMode: `true`
> Reading article

pon1

> privacyMode: `false`
> In portal
> {portalName}
> <button>Enter the portal</button>
poff2

> privacyMode: `true`
> In portal
pon2

File: websites/#/Фугапедия/presence.ts [hunk 1]
```diff
@@ -3,79 +3,129 @@ const presence = new Presence({
...
   else if (path === "/book") {
-    const bookName = document.querySelector("p.text-4xl.font-bold.font-3").textContent;
     presenceData.details = "Читает книгу";
-    presenceData.state = bookName;
-    presenceData.buttons = [
-      {
-        label: "Перейти к книге",
-        url: document.URL
+    if (!privacyMode) {
+      const bookName = document.querySelector("p.text-4xl.font-bold.font-3").textContent;
```
Comment (EncryptedDev) [inline websites/#/Фугапедия/presence.ts:102]: Variable is only used once

File: websites/#/Фугапедия/presence.ts [hunk 2]
```diff
@@ -3,79 +3,129 @@ const presence = new Presence({
...
   } else if (path === "/portal") {
-    const portalName = document.querySelector("p.text-5xl.font-bold.text-black.place-self-center.ml-8").textContent;
     presenceData.details = "В портале";
-    presenceData.state = portalName;
-    presenceData.buttons = [
-      {
-        label: "Войти в портал",
-        url: document.URL
+    if (!privacyMode) {
+      const portalName = document.querySelector("p.text-4xl.font-semibold.text-white").textContent;
```
Comment (EncryptedDev) [inline websites/#/Фугапедия/presence.ts:87]: Variable is only used once

File: websites/#/Фугапедия/presence.ts [hunk 3]
```diff
@@ -3,79 +3,129 @@ const presence = new Presence({
...
+    presenceData.details = "Смотрит список правок";
+    presenceData.state = "на всей вики";
   } else if (path === "/history") {
-    const pageTitle = document.querySelector("title").textContent,
-      articleTitle = pageTitle.slice(2, pageTitle.length - 12);
     presenceData.details = "Смотрит историю статьи";
-    presenceData.state = articleTitle;
+    if (!privacyMode) {
+      const pageTitle = document.querySelector("title").textContent,
+        articleTitle = pageTitle.slice(2, pageTitle.length - 12);
```
Comment (EncryptedDev) [inline websites/#/Фугапедия/presence.ts:81]: Variable is only used once

File: websites/#/Фугапедия/presence.ts [hunk 4]
```diff
@@ -3,79 +3,129 @@ const presence = new Presence({
...
   } else if (path === "/search") presenceData.details = "В поисках статьи";
   else if (path === "/create") presenceData.details = "Создаёт статью";
   else if (path === "/edit") {
-    const pageTitle = document.querySelector("title").textContent,
-      articleTitle = pageTitle.slice(2, pageTitle.length - 12);
     presenceData.details = "Редактирует статью";
-    presenceData.state = articleTitle;
+    if (!privacyMode) {
+      const pageTitle = document.querySelector("title").textContent,
+        articleTitle = pageTitle.slice(2, pageTitle.length - 12);
```
Comment (EncryptedDev) [inline websites/#/Фугапедия/presence.ts:71]: Variable is only used once

File: websites/#/Фугапедия/presence.ts [hunk 5]
```diff
@@ -3,79 +3,129 @@ const presence = new Presence({
...
   } else if (path === "/category") {
-    const categoryName = document.querySelector("p.text-4xl.font-bold.font-3").textContent;
     presenceData.details = "Смотрит категорию";
-    presenceData.state = categoryName;
-    presenceData.buttons = [
-      {
-        label: "Перейти в категорию",
-        url: document.URL
+    if (!privacyMode) {
+      const categoryName = document.querySelector("p.text-4xl.font-bold.font-3").textContent;
```
Comment (EncryptedDev) [inline websites/#/Фугапедия/presence.ts:54]: Variable is only used once

File: websites/#/Фугапедия/presence.ts [hunk 6]
```diff
@@ -3,79 +3,129 @@ const presence = new Presence({
...
+            url: document.URL
+          }
+        ];
       }
-    ];
+    }
+  } else if (path === "/view_c") {
+    presenceData.details = "Смотрит правку";
+    if (!privacyMode) {
+      const editTitle = document.querySelector("p.text-4xl.font-bold.font-3").textContent;
```
Comment (EncryptedDev) [inline websites/#/Фугапедия/presence.ts:40]: Variable is only used once

File: websites/#/Фугапедия/presence.ts [hunk 7]
```diff
@@ -3,79 +3,129 @@ const presence = new Presence({
...
   else if (path === "/view") {
-    const articleTitle = document.querySelector("p.text-4xl.font-bold.font-3").textContent;
     presenceData.details = "Читает статью";
-    presenceData.state = articleTitle;
-    presenceData.buttons = [
-      {
-        label: "Перейти к статье",
-        url: document.URL
+    if (!privacyMode) {
+      const articleTitle = document.querySelector("p.text-4xl.font-bold.font-3").textContent;
```
Comment (EncryptedDev) [inline websites/#/Фугапедия/presence.ts:26]: Variable is only used once

File: websites/#/Фугапедия/presence.ts [hunk 8]
```diff
@@ -3,79 +3,129 @@ const presence = new Presence({
...
-      name = document.querySelector("p.text-lg.text-gray-600.ml-4.place-self-center").textContent;
     presenceData.details = "Смотрит профиль";
-    presenceData.state = `${nickname} (${name})`;
-    presenceData.buttons = [
-      {
-        label: "Перейти к пользователю",
-        url: document.URL
+    if (!privacyMode) {
+      const nickname = document.querySelector("p.text-2xl.text-black.font-medium.ml-4.place-self-center").firstChild.textContent,
+        name = document.querySelector("p.text-lg.text-gray-600.ml-4.place-self-center").textContent;
```
Comment (EncryptedDev) [inline websites/#/Фугапедия/presence.ts:118]: Variables are only used once

**Review by EncryptedDev — DISMISSED** [review]

**Comment by Syjalo** [comment]
If I'll remove the variables I'll lose information about what `querySelector` returns and I'll have to insert comments. As for me, variables look better than comments.

**Review by Slowlife01 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**